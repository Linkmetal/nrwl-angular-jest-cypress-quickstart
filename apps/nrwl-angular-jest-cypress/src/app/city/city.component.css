.city {
    font: 200 20px/1.5 Helvetica, Verdana, sans-serif;
    position: fixed; 
    top: 20px; 



    right: 0;
    width: 300px;
    min-height: 100px;
    background: hsla(80, 90%, 40%, 0.7);
    color: white;
    font-weight: bold;
    font-size: large;
    text-align: left;
    border: solid hsla(80, 90%, 40%, 0.5);
    border-right: none;
    padding: 0.5em 0.5em 0.5em 2.5em;
    box-shadow: 0 1px 3px black;
    border-radius: 3em 0.5em 0.5em 3em;
}