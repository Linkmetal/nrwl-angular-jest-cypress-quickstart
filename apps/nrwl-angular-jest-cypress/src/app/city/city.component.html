<div style="position: fixed; top: 0; right: 20px;">
  <h1 *ngIf="cityService.getCity() | async as city">
    {{ city }}
  </h1>
</div>

