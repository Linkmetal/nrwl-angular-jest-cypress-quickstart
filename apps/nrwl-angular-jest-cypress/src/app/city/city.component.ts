import { Component, OnInit } from '@angular/core';
import { CityService } from '../city.service';

@Component({
  selector: 'nrwl-angular-jest-cypress-city',
  templateUrl: './city.component.html',
  styleUrls: ['./city.component.css']
})
export class CityComponent {
  constructor(public cityService: CityService) { }
}
