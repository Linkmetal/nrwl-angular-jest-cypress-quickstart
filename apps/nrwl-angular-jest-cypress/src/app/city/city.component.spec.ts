import { async, ComponentFixture, TestBed } from '@angular/core/testing';

import { CityComponent } from './city.component';
import { CityService } from '../city.service';
import { of } from 'rxjs';

jest.mock('../city.service');

describe('CityComponent', () => {
  let component: CityComponent;
  let fixture: ComponentFixture<CityComponent>;

  let cityService: CityService;

  beforeEach(async(() => {
    TestBed.configureTestingModule({
      declarations: [ CityComponent ],
      providers: [ CityService ]
    })
    .compileComponents();
  }));

  beforeEach(() => {
    cityService = TestBed.get(CityService);

    fixture = TestBed.createComponent(CityComponent);
    component = fixture.componentInstance;

    jest.spyOn(cityService, 'getCity').mockReturnValue(of('Madrid'));

    fixture.detectChanges();
  });

  it('should create', () => {
    expect(component).toBeTruthy();
  });

  it('should render component properly', () => {
    expect(fixture).toMatchSnapshot();
  });
});
