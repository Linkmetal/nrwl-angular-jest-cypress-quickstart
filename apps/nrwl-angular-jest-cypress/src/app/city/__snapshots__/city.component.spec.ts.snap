// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CityComponent should render component properly 1`] = `
<nrwl-angular-jest-cypress-city
  cityService={[Function CityService]}
>
  <div
    style="position: fixed; top: 0; right: 20px;"
  >
    
    <h1>
       Madrid 
    </h1>
  </div>
</nrwl-angular-jest-cypress-city>
`;
